kernel/arch/dreamcast/kernel/init.c

```diff
@@ -280,7 +281,17 @@ void arch_shutdown() {
 }
 
 /* Generic kernel exit point (configurable) */
-void arch_exit() {
+__noreturn void arch_exit() {
+    exit(0);
+}
+
+/* Return point from newlib's _exit() */
+__noreturn void post_newlib_exit(int ret_code) {
+    dbglog(DBG_INFO, "arch: exit value %d\n", ret_code);
```

Comment:
I like that you're explicitly printing this, but maybe it should be like "arch: exit return code: %d", to make it more clear where the value is coming from, what it represents, and how you can change it.